<a href="https://colab.research.google.com/github/asthanas/DataScienceProjects/blob/master/Worlds_Wealthiest.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **World's Wealthiest - Descriptive Statistics**

The World's Billionaires is an annual ranking by documented net worth of the world's wealthiest billionaires compiled and published in March annually by the American business magazine Forbes. The list was first published in March 1987. The total net worth of each individual on the list is estimated and is cited in United States dollars, based on their documented assets and accounting for debt. Royalty and dictators whose wealth comes from their positions are excluded from these lists. This ranking is an index of the wealthiest documented individuals, excluding and ranking against those with wealth that is not able to be completely ascertained. (wikipedia)

The dataset has following features:
 - Year
 - Rank
 - Name
 - Net_Worth
 - Age
 - Nationality
 - Source_wealth

**Objective**: Perform descriptive analytics to understand what data tells us about the world's wealthiest

In [0]:
import numpy as np
import pandas as pd
import scipy
from scipy import stats
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
pd.set_option('display.max_columns',25)

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


Read input file billionaires.csv

In [0]:
data=pd.read_csv("billionaires.csv")

**Read top 10 values from the csv file**

In [0]:
data.head(10)

year  rank               name net_worth  age     natinality  \
0  2019     1         Jeff Bezos      131    55  United States   
1  2019     2         Bill Gates      96.5   63  United States   
2  2019     3     Warren Buffett      82.5   88  United States   
3  2019     4    Bernard Arnault       76    70         France   
4  2019     5        Carlos Slim       64    79         Mexico   
5  2019     6     Amancio Ortega      62.7   82          Spain   
6  2019     7      Larry Ellison      62.5   74  United States   
7  2019     8    Mark Zuckerberg      62.3   34  United States   
8  2019     9  Michael Bloomberg      55.5   77  United States   
9  2019    10         Larry Page      50.8   45  United States   

                source_wealth  
0                      Amazon  
1                   Microsoft  
2          Berkshire Hathaway  
3                        LVMH  
4  América Móvil, Grupo Carso  
5               Inditex, Zara  
6          Oracle Corporation  
7                    Facebook  
8              Bloomberg L.P.  
9               Alphabet Inc.

Correct the column name from "natinality" to "nationality"

In [0]:
data.rename(columns = {'natinality':'nationality'}, inplace = True)

In [0]:
data.head(20)

year  rank               name net_worth  age    nationality  \
0   2019     1         Jeff Bezos      131    55  United States   
1   2019     2         Bill Gates      96.5   63  United States   
2   2019     3     Warren Buffett      82.5   88  United States   
3   2019     4    Bernard Arnault       76    70         France   
4   2019     5        Carlos Slim       64    79         Mexico   
5   2019     6     Amancio Ortega      62.7   82          Spain   
6   2019     7      Larry Ellison      62.5   74  United States   
7   2019     8    Mark Zuckerberg      62.3   34  United States   
8   2019     9  Michael Bloomberg      55.5   77  United States   
9   2019    10         Larry Page      50.8   45  United States   
10  2018     1         Jeff Bezos      112    54  United States   
11  2018     2         Bill Gates       90    62  United States   
12  2018     3     Warren Buffett       84    87  United States   
13  2018     4    Bernard Arnault       72    69         France   
14  2018     5    Mark Zuckerberg       71    33  United States   
15  2018     6     Amancio Ortega       70    81          Spain   
16  2018     7        Carlos Slim      67.1   78         Mexico   
17  2018     8       Charles Koch       60    82  United States   
18  2018     8         David Koch       60    77  United States   
19  2018    10      Larry Ellison      58.5   73  United States   

                 source_wealth  
0                       Amazon  
1                    Microsoft  
2           Berkshire Hathaway  
3                         LVMH  
4   América Móvil, Grupo Carso  
5                Inditex, Zara  
6           Oracle Corporation  
7                     Facebook  
8               Bloomberg L.P.  
9                Alphabet Inc.  
10                      Amazon  
11                   Microsoft  
12          Berkshire Hathaway  
13                        LVMH  
14                    Facebook  
15               Inditex, Zara  
16  América Móvil, Grupo Carso  
17             Koch Industries  
18             Koch Industries  
19          Oracle Corporation

In [0]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   year           180 non-null    int64 
 1   rank           180 non-null    int64 
 2   name           180 non-null    object
 3   net_worth      180 non-null    object
 4   age            180 non-null    int64 
 5   nationality    180 non-null    object
 6   source_wealth  180 non-null    object
dtypes: int64(3), object(4)
memory usage: 10.0+ KB


In [0]:
data.describe()

year        rank         age
count   180.000000  180.000000  180.000000
mean   2010.500000    5.394444   66.666667
std       5.202599    2.793642   13.689698
min    2002.000000    1.000000   31.000000
25%    2006.000000    3.000000   55.000000
50%    2010.500000    5.500000   69.000000
75%    2015.000000    8.000000   78.000000
max    2019.000000   10.000000   92.000000

In [0]:
data.describe(include='object')

name net_worth    nationality source_wealth
count          180       180            180           180
unique          45       122             14            43
top     Bill Gates      20.0  United States     Microsoft
freq            18         9             99            23

In [0]:
data['net_worth']=data['net_worth'].astype('float')

In [0]:
data['year']=data['year'].astype('object')

In [0]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   year           180 non-null    object 
 1   rank           180 non-null    int64  
 2   name           180 non-null    object 
 3   net_worth      180 non-null    float64
 4   age            180 non-null    int64  
 5   nationality    180 non-null    object 
 6   source_wealth  180 non-null    object 
dtypes: float64(1), int64(2), object(4)
memory usage: 10.0+ KB


In [0]:
data['source_wealth'].value_counts()

Microsoft                                  23
Berkshire Hathaway                         18
Oracle Corporation                         14
Koch Industries                            12
Wal-Mart                                   11
Aldi Süd                                    7
Walmart                                     6
Telmex, América Móvil, Grupo Carso          6
Inditex Group                               6
Arcelor Mittal                              5
IKEA                                        5
Reliance Industries                         4
Amazon                                      4
Kingdom Holding Company                     4
LVMH Moët Hennessy • Louis Vuitton          4
Facebook                                    4
LVMH                                        4
América Móvil, Grupo Carso                  4
Inditex, Zara                               3
EBX Group                                   3
Inditex                                     3
Bloomberg L.P.                    

Based on the above count we can say tha worlds wealthiest people belong to microsoft

In [0]:
data_by_year=data.groupby('year')
data_by_year.describe()

rank                                            net_worth         \
     count mean       std  min   25%  50%   75%   max     count   mean   
year                                                                     
2002  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  27.52   
2003  10.0  5.2  2.616189  1.0  3.25  5.5  7.75   8.0      10.0  26.63   
2004  10.0  4.5  1.900292  1.0  3.25  5.5  6.00   6.0      10.0  25.50   
2005  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  26.22   
2006  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  27.74   
2007  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  34.35   
2008  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  42.60   
2009  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  25.39   
2010  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  34.22   
2011  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  40.61   
2012  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  39.54   
2013  10.0  5.4  2.988868  1.0  3.25  5.5  7.50  10.0      10.0  45.15   
2014  10.0  5.4  2.988868  1.0  3.25  5.5  7.50  10.0      10.0  50.76   
2015  10.0  5.4  2.988868  1.0  3.25  5.5  7.50  10.0      10.0  55.60   
2016  10.0  5.4  2.875181  1.0  3.25  5.5  7.75   9.0      10.0  50.54   
2017  10.0  5.4  2.913570  1.0  3.25  5.5  7.75  10.0      10.0  61.25   
2018  10.0  5.4  2.913570  1.0  3.25  5.5  7.75  10.0      10.0  74.46   
2019  10.0  5.5  3.027650  1.0  3.25  5.5  7.75  10.0      10.0  74.38   

                                                      age                   \
            std   min     25%    50%     75%    max count  mean        std   
year                                                                         
2002  14.434665  16.1  19.550  20.00  27.400   60.0  10.0  55.0  15.570628   
2003  10.266889  20.5  20.550  22.15  26.400   52.8  10.0  60.8  12.968338   
2004  10.229152  20.0  20.000  20.50  22.625   46.6  10.0  61.6  13.882043   
2005  10.339761  18.3  19.125  23.35  24.700   46.5  10.0  62.8  12.787146   
2006  11.020304  18.8  20.375  22.75  29.500   52.0  10.0  64.3  12.970479   
2007  13.004380  22.0  24.500  29.25  45.000   56.0  10.0  65.9  11.512795   
2008  13.631662  27.0  30.250  42.50  54.750   62.0  10.0  63.8  16.287691   
2009   8.439648  18.3  19.350  21.75  31.875   40.0  10.0  70.6  13.672357   
2010  11.932011  23.5  27.125  28.35  42.500   53.5  10.0  66.0  12.328828   
2011  15.351399  26.5  30.250  35.30  47.750   74.0  10.0  63.7   9.031427   
2012  15.050079  25.4  27.000  36.75  43.250   69.0  10.0  70.8  12.072006   
2013  16.353814  29.0  31.750  38.50  56.125   73.0  10.0  74.2   9.908470   
2014  15.550791  34.7  38.500  44.00  62.550   76.0  10.0  72.3   7.803845   
2015  16.239732  40.1  42.000  48.60  70.650   79.2  10.0  74.3   9.626930   
2016  12.636824  39.6  40.900  44.90  58.100   75.0  10.0  68.3  16.330272   
2017  13.873976  47.5  49.275  55.25  72.425   86.0  10.0  69.3  16.330272   
2018  16.689132  58.5  61.775  70.50  81.000  112.0  10.0  69.6  16.208365   
2019  24.054050  50.8  62.350  63.35  80.875  131.0  10.0  66.7  17.320829   

                                      
       min    25%   50%    75%   max  
year                                  
2002  35.0  43.25  51.0  66.00  80.0  
2003  46.0  53.25  56.5  67.75  82.0  
2004  47.0  52.00  57.0  69.75  84.0  
2005  49.0  52.50  60.5  71.75  85.0  
2006  49.0  53.50  61.5  76.50  82.0  
2007  49.0  56.50  69.0  75.25  80.0  
2008  40.0  51.25  62.5  76.75  88.0  
2009  52.0  59.50  71.0  81.75  89.0  
2010  53.0  55.50  63.5  73.00  90.0  
2011  53.0  56.25  62.0  69.75  80.0  
2012  55.0  63.25  69.5  79.50  92.0  
2013  57.0  69.00  74.5  80.75  90.0  
2014  58.0  66.25  73.5  77.75  83.0  
2015  59.0  67.00  74.5  78.75  92.0  
2016  31.0  62.75  74.5  78.25  85.0  
2017  32.0  63.75  75.5  79.25  86.0  
2018  33.0  63.75  75.0  80.25  87.0  
2019  34.0  57.00  72.0  78.50  

Describe funtion in python is a powerful when trying to get statistical details of entire data, as above we can see mean, median and mode details of each column.
For 2002 mean age of billionair is 55 and in 2019 mean age is 66.7, 
For 2002 mean networth of billionair is 27.53 and in 2019 mean networth is 74.38

**Descriptive statistics** is the summary given data set. Descriptive statistics are broken down into 1. Measures of Central Tendency and 2. Measures of Spread

# **Measures of Central Tendency**

The obvious question when looking at a salary dataset is "How much do people make?". And when asking that nobody is interested to get 100s of rows of data. They want just a single number which can represent the entire dataset. And that's exactly what Central Tendency seeks to do. There are three measures of central tendency viz. **Mean, Median, Mode**



**MEAN**: The mean is the average value.

In [0]:
data_by_year[['net_worth','age']].mean()

net_worth   age
year                 
2002      27.52  55.0
2003      26.63  60.8
2004      25.50  61.6
2005      26.22  62.8
2006      27.74  64.3
2007      34.35  65.9
2008      42.60  63.8
2009      25.39  70.6
2010      34.22  66.0
2011      40.61  63.7
2012      39.54  70.8
2013      45.15  74.2
2014      50.76  72.3
2015      55.60  74.3
2016      50.54  68.3
2017      61.25  69.3
2018      74.46  69.6
2019      74.38  66.7

**Observation** : Mean net_worth and age for each year